main.go

```diff
@@ -390,22 +392,42 @@ func SaveBackupAction(c *cli.Context) error {
 					"name":  backupName,
 					"error": err,
 				}).Warn("Error while trying to retrieve cluster state from cluster")
-
 			}
-			if err = CreateBackup(backupName, etcdCACert, etcdCert, etcdKey, etcdEndpoints, client, bc); err == nil {
+			compressedFilePath, err := CreateBackup(backupName, etcdCACert, etcdCert, etcdKey, etcdEndpoints, backupRetries, s3Retries)
+			if err == nil {
 				DeleteBackups(backupTime, retentionPeriod)
-				if s3Backup {
-					DeleteS3Backups(backupTime, retentionPeriod, client, bc)
+				if !bc.Backup {
+					continue
 				}
+				err = CreateS3Backup(backupName, compressedFilePath, bc)
+				if err != nil {
+					continue
+				}
+				DeleteS3Backups(backupTime, retentionPeriod, bc)
 			}
 		}
 	}
 }
 
-func CreateBackup(backupName string, etcdCACert, etcdCert, etcdKey, endpoints string, svc *minio.Client, server *backupConfig) error {
+func minioClientFromConfig(bc *backupConfig) (*minio.Client, error) {
+	client, err := setS3Service(bc, true)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"s3-endpoint":    bc.Endpoint,
+			"s3-bucketName":  bc.BucketName,
+			"s3-accessKey":   bc.AccessKey,
+			"s3-region":      bc.Region,
+			"s3-endpoint-ca": bc.EndpointCA,
+			"s3-folder":      bc.Folder,
+		}).Errorf("failed to set s3 server: %s", err)
+		return nil, fmt.Errorf("failed to set s3 server: %+v", err)
+	}
+	return client, nil
+}
+
+func CreateBackup(backupName, etcdCACert, etcdCert, etcdKey, endpoints string, backupRetries, s3Retries int) (compressedFilePath string, err error) {
```

Comment:
the parameter`s3Retries` is not used in the function

Reply:
Another good catch!